### ISSUE TITLE
npm ERR! Unexpected token '.' with  with nvm-windows <= 1.1.7

### ISSUE THREAD

**Issue by syedbilal07** [npm/cli#4234]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

I have updated to the latest npm v8.3.1 and it is showing me an error while using the command "npm outdated -g --depth=0". See the image; https://imgur.com/a/5fMZlye. It was fine on v8.3.0. What is the solution?

### Expected Behavior

It should return the outdated packages after a successful run. But it is giving error. 

### Steps To Reproduce

1. In this environment...
2. With this config...
3. Run '...'
4. See error...
I am using Windows 10 (Enterprise - Latest Version). I have the latest node and npm installed at the moment.

### Environment

- npm:
- Node.js:
- OS Name:
- System Model Name:
- npm config:
```ini
; copy and paste output from `npm config ls` here
```
; "user" config from C:\Users\Bilal\.npmrc

registry = "http://registry.npmjs.org/"

; node bin location = C:\Program Files\nodejs\node.exe
; cwd = C:\Users\Bilal
; HOME = C:\Users\Bilal
; Run `npm config ls -l` to show all defaults.

**Comment by syedbilal07** [comment]
27 verbose stack SyntaxError: Unexpected token '.'
27 verbose stack     at Object.compileFunction (node:vm:352:18)
27 verbose stack     at wrapSafe (node:internal/modules/cjs/loader:1026:15)
27 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1061:27)
27 verbose stack     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1149:10)
27 verbose stack     at Module.load (node:internal/modules/cjs/loader:975:32)
27 verbose stack     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
27 verbose stack     at Module.require (node:internal/modules/cjs/loader:999:19)
27 verbose stack     at require (node:internal/modules/cjs/helpers:102:18)
27 verbose stack     at Object.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\commands\outdated.js:11:18)
27 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1097:14)

**Comment by syedbilal07** [comment]
Fixed it on my own. Please replace it with my version https://files.fm/u/sz3yrgsvj

**Comment by dcodeIO** [comment]
Seeing this issue on GitHub actions as well. Seems to be isolated to "windows" images running Node.js 17:

https://github.com/AssemblyScript/assemblyscript/runs/4856320026?check_suite_focus=true#step:4:4

OS: Microsoft Windows Server 2019 10.0.17763
Node.js: 17.4.0
npm: 8.3.1

**Comment by syedbilal07** [comment]
Facing this issue on  npm install script too.

**Comment by syedbilal07** [comment]
27 verbose stack ../workspaces/libnpmfund
27 verbose stack ^
27 verbose stack
27 verbose stack SyntaxError: Unexpected token '.'

**Comment by bonesoul** [comment]
same here..

**Comment by bonesoul** [comment]
any fixes for this?

**Comment by syedbilal07** [comment]
Not yet. Waiting for the community member to resolve this

**Comment by ruyadorno** [comment]
hi @syedbilal07 it would be interesting to look at the log files to see understand better what could have happened there, if that's something you can provide us. (or anyone else having this issue)

**Comment by UziTech** [comment]
I don't know if there is a way to look at the logs in github actions but here is one that is failing. https://github.com/coffeelint/coffeelint/runs/4960440881

```
Run npm ci
  npm ci
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
npm ERR! Unexpected token '.'

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\npm\cache\_logs\2022-01-27T01_47_31_169Z-debug-0.log
Error: Process completed with exit code 1.
```

**Comment by halcwb** [comment]
The log files for the error:

<details>
    <summary>log file</summary>
0 verbose cli [
0 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
0 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
0 verbose cli   'install'
0 verbose cli ]
1 info using npm@8.3.1
2 info using node@v17.4.0
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 2ms
5 timing config:load:file:C:\Users\halcw\AppData\Roaming\nvm\v17.4.0\node_modules\npm\npmrc Completed in 24ms
6 timing config:load:builtin Completed in 24ms
7 timing config:load:cli Completed in 2ms
8 timing config:load:env Completed in 0ms
9 timing config:load:file:C:\Development\halcwb\apps\GenPres2\.npmrc Completed in 0ms
10 timing config:load:project Completed in 3ms
11 timing config:load:file:C:\Users\halcw\.npmrc Completed in 0ms
12 timing config:load:user Completed in 0ms
13 timing config:load:file:C:\Program Files\nodejs\etc\npmrc Completed in 1ms
14 timing config:load:global Completed in 1ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:credentials Completed in 1ms
17 timing config:load:setEnvs Completed in 1ms
18 timing config:load Completed in 36ms
19 timing npm:load:configload Completed in 36ms
20 timing npm:load:setTitle Completed in 1ms
21 timing config:load:flatten Completed in 3ms
22 timing npm:load:display Completed in 5ms
23 verbose logfile C:\Users\halcw\AppData\Local\npm-cache\_logs\2022-01-26T17_31_57_523Z-debug-0.log
24 timing npm:load:logFile Completed in 52ms
25 timing npm:load:timers Completed in 0ms
26 timing npm:load:configScope Completed in 0ms
27 timing npm:load Completed in 94ms
28 verbose stack C:\Users\halcw\AppData\Roaming\nvm\v17.4.0\node_modules\npm\node_modules\libnpmfund:1
28 verbose stack ../workspaces/libnpmfund
28 verbose stack ^
28 verbose stack
28 verbose stack SyntaxError: Unexpected token '.'
28 verbose stack     at Object.compileFunction (node:vm:352:18)
28 verbose stack     at wrapSafe (node:internal/modules/cjs/loader:1026:15)
28 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1061:27)
28 verbose stack     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1149:10)
28 verbose stack     at Module.load (node:internal/modules/cjs/loader:975:32)
28 verbose stack     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
28 verbose stack     at Module.require (node:internal/modules/cjs/loader:999:19)
28 verbose stack     at require (node:internal/modules/cjs/helpers:102:18)
28 verbose stack     at Object.<anonymous> (C:\Users\halcw\AppData\Roaming\nvm\v17.4.0\node_modules\npm\lib\utils\reify-output.js:16:38)
28 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1097:14)
29 verbose cwd C:\Development\halcwb\apps\GenPres2
30 verbose Windows_NT 10.0.19043
31 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
32 verbose node v17.4.0
33 verbose npm  v8.3.1
34 error Unexpected token '.'
35 verbose exit 1
36 timing npm Completed in 308ms
37 verbose code 1
38 error A complete log of this run can be found in:
38 error     C:\Users\halcw\AppData\Local\npm-cache\_logs\2022-01-26T17_31_57_523Z-debug-0.log
</details>

On my machine:
Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19043
 OS Platform: Windows
 RID:         win10-x64

Luckily I use nvm so I could easily revert back to node 17.3.1.

**Comment by syedbilal07** [comment]
Hi @ruyadorno, I am now using npm v8.3.0 with node 17.4.0. I cannot provide the full log right now and I do not want to upgrade to npm v8.3.2. Please check the logs on your end and run tests on Github so you can figure it out. I have provided stack trace of the issue above. I think that could help.

**Comment by feeldiac** [comment]
I had a similar issue when using `npm init` and `npm install`. I downgrade my node version to 16.13.2 (recommended for most users) and it works.

**Comment by RodEsp** [comment]
Heya @syedbilal07 was this actually fixed or did you just close it because you're using your own workaround?

**Comment by bonesoul** [comment]
use the newest version of npm and u should be fine. just skip 8.3.1.

**Comment by kernju** [comment]
I also have the same issue, using nvm for Windows (Node 17.4.0, npm 8.3.1).

> 0 verbose cli [
0 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
0 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
0 verbose cli   'install',
0 verbose cli   '-g',
0 verbose cli   '@angular/cli'
0 verbose cli ]
1 info using npm@8.3.1
2 info using node@v17.4.0
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 1ms
5 timing config:load:file:C:\Users\xxx\AppData\Roaming\nvm\v17.4.0\node_modules\npm\npmrc Completed in 2ms
6 timing config:load:builtin Completed in 2ms
7 timing config:load:cli Completed in 2ms
8 timing config:load:env Completed in 0ms
9 timing config:load:project Completed in 3ms
10 timing config:load:file:C:\Users\xxx\.npmrc Completed in 1ms
11 timing config:load:user Completed in 1ms
12 timing config:load:file:C:\Program Files\nodejs\etc\npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:validate Completed in 1ms
15 timing config:load:credentials Completed in 1ms
16 timing config:load:setEnvs Completed in 1ms
17 timing config:load Completed in 12ms
18 timing npm:load:configload Completed in 12ms
19 timing npm:load:setTitle Completed in 0ms
20 timing config:load:flatten Completed in 2ms
21 timing npm:load:display Completed in 3ms
22 verbose logfile C:\Users\xxx\AppData\Local\npm-cache\_logs\2022-02-04T14_14_48_111Z-debug-0.log
23 timing npm:load:logFile Completed in 13ms
24 timing npm:load:timers Completed in 0ms
25 timing npm:load:configScope Completed in 0ms
26 timing npm:load Completed in 30ms
27 verbose stack C:\Users\xxx\AppData\Roaming\nvm\v17.4.0\node_modules\npm\node_modules\libnpmfund:1
27 verbose stack ../workspaces/libnpmfund
27 verbose stack ^
27 verbose stack
27 verbose stack SyntaxError: Unexpected token '.'
27 verbose stack     at Object.compileFunction (node:vm:352:18)
27 verbose stack     at wrapSafe (node:internal/modules/cjs/loader:1026:15)
27 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1061:27)
27 verbose stack     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1149:10)
27 verbose stack     at Module.load (node:internal/modules/cjs/loader:975:32)
27 verbose stack     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
27 verbose stack     at Module.require (node:internal/modules/cjs/loader:999:19)
27 verbose stack     at require (node:internal/modules/cjs/helpers:102:18)
27 verbose stack     at Object.<anonymous> (C:\Users\xxx\AppData\Roaming\nvm\v17.4.0\node_modules\npm\lib\utils\reify-output.js:16:38)
27 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1097:14)
28 verbose cwd C:\Program Files\nodejs
29 verbose Windows_NT 10.0.19043
30 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "@angular/cli"
31 verbose node v17.4.0
32 verbose npm  v8.3.1
33 error Unexpected token '.'
34 verbose exit 1
35 timing npm Completed in 208ms
36 verbose code 1
37 error A complete log of this run can be found in:
37 error     C:\Users\xxx\AppData\Local\npm-cache\_logs\2022-02-04T14_14_48_111Z-debug-0.log

**Comment by shubhambattoo** [comment]
> use the newest version of npm and u should be fine. just skip 8.3.1.

how to do that? to update npm you need npm.

**Comment by kernju** [comment]
I finally used the latest Node.js LTS, 16.3.2, and it worked.

**Comment by cristiand391** [comment]
Yesterday node v16.14.0 (latest LTS) was released, and it includes npm v8.3.1 so there's no way to make npm update itself.
If you see this error when running latest LTS on windows, you might want to install previous LTS v16.13.2 👍🏼

**Comment by gfieremans** [comment]
> > use the newest version of npm and u should be fine. just skip 8.3.1.
> 
> how to do that? to update npm you need npm.

Just ran in the same issue with node 16.14.0. You can download 8.3.2 [tgz file](https://registry.npmjs.org/npm/-/npm-8.3.2.tgz) and manually replace npm in your node_modules directory of the node version you have.

**Comment by UziTech** [comment]
Still happens with npm 8.4.1

https://github.com/UziTech/tonybrix/runs/5150397283?check_suite_focus=true

**Comment by bonesoul** [comment]
i can also confirm that this still exists with 8.4.1.

**Comment by bonesoul** [comment]
further pushed the discussion to node guys to here: https://github.com/nodejs/node/pull/41836

**Comment by MylesBorins** [comment]
Can anyone confirm if this is fixed in v8.5.0

**Comment by cristiand391** [comment]
@MylesBorins I wasn't able to reproduce this error using the official installers.

**node v17.5.0 / npm v8.4.1**
![Screenshot (3)](https://user-images.githubusercontent.com/6853656/153725802-9c817b4f-9f76-4d02-9202-6b84f916c221.png)
**node v16.14.0 / npm v8.3.1**
![Screenshot (4)](https://user-images.githubusercontent.com/6853656/153725803-15e82fd9-1e53-482c-a4e1-e3dc809dd6dd.png)

Some users reported that the error only happens when using an old version of `nvm-windows`: https://github.com/npm/cli/issues/4340

I installed latest `nvm-windows` v1.1.9 and npm v8.3.1 works as expected. I didn't try an old version of nvm-windows on my machine but can confirm that the Windows image we use has `NVM (Node Version Manager) 1.1.7` pre-installed: https://circleci.com/docs/2.0/hello-world-windows/#software-pre-installed-in-the-windows-image

You can see one of our windows workflows using it to install latest node LTS v16.14.0 here: https://app.circleci.com/pipelines/github/salesforcecli/plugin-org/2412/workflows/590d8ff9-a2dd-4a81-9cc4-3295a62939e2/jobs/7193

**Comment by bonesoul** [comment]
well i do use nvm-windows and will update it to see if it's fixed.

**Comment by timothystewart6** [comment]
I am using the latest `nvm-windows`  and I cannot install any packages using node `16.14.0`  with npm `8.3.1` which was installed via `nvm-windows`


```log
npm install -g pm2
npm ERR! Unexpected token '.'
```

**Comment by cristiand391** [comment]
@timothystewart6 what does `nvm version` returns? Can you try uninstalling nvm and use the official installer to see if the error shows up?

**Comment by learningnewthings** [comment]
Same as @timothystewart6 , even npm init returns npm ERR! Unexpected token '.'
@cristiand391 - this was with both nvm 1.1.7 and nvm 1.1.9

**Comment by syedbilal07** [comment]
Can someone confirm me if it is working with Node 17.5.0 and npm 8.5.0? Or the issue is still present?

**Comment by bonesoul** [comment]
we need a clean install of nvm-windows and remove old version of already installed node. then it's all fine.

**Comment by janpio** [comment]
(Just leaving a comment that I ended up here as well via using `nvm-windows` to install the most recent Node ` 17.5.0` (32 bit) which comes with Npm `8.3.1`. Installing an older Node, in my case `16.13.0 32`, worked to unblock me.)

**Comment by andrey-helldar** [comment]
Node 16.14.0
Npm 8.3.1

Same error

```
0 verbose cli [
0 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
0 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
0 verbose cli   'i'
0 verbose cli ]
1 info using npm@8.3.1
2 info using node@v16.14.0
3 timing npm:load:whichnode Completed in 1ms
4 timing config:load:defaults Completed in 2ms
5 timing config:load:file:C:\Users\Helldar\AppData\Roaming\nvm\v16.14.0\node_modules\npm\npmrc Completed in 3ms
6 timing config:load:builtin Completed in 4ms
7 timing config:load:cli Completed in 3ms
8 timing config:load:env Completed in 2ms
9 timing config:load:file:C:\dev\domains\crm\frontend\.npmrc Completed in 1ms
10 timing config:load:project Completed in 7ms
11 timing config:load:file:C:\Users\Helldar\.npmrc Completed in 1ms
12 timing config:load:user Completed in 2ms
13 timing config:load:file:C:\Program Files\nodejs\etc\npmrc Completed in 0ms
14 timing config:load:global Completed in 0ms
15 timing config:load:validate Completed in 2ms
16 timing config:load:credentials Completed in 0ms
17 timing config:load:setEnvs Completed in 1ms
18 timing config:load Completed in 25ms
19 timing npm:load:configload Completed in 25ms
20 timing npm:load:setTitle Completed in 0ms
21 timing config:load:flatten Completed in 3ms
22 timing npm:load:display Completed in 6ms
23 verbose logfile C:\Users\Helldar\AppData\Local\npm-cache\_logs\2022-02-14T13_39_55_054Z-debug-0.log
24 timing npm:load:logFile Completed in 14ms
25 timing npm:load:timers Completed in 0ms
26 timing npm:load:configScope Completed in 0ms
27 timing npm:load Completed in 46ms
28 verbose stack C:\Users\Helldar\AppData\Roaming\nvm\v16.14.0\node_modules\npm\node_modules\libnpmfund:1
28 verbose stack ../workspaces/libnpmfund
28 verbose stack ^
28 verbose stack
28 verbose stack SyntaxError: Unexpected token '.'
28 verbose stack     at Object.compileFunction (node:vm:352:18)
28 verbose stack     at wrapSafe (node:internal/modules/cjs/loader:1032:15)
28 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1067:27)
28 verbose stack     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
28 verbose stack     at Module.load (node:internal/modules/cjs/loader:981:32)
28 verbose stack     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
28 verbose stack     at Module.require (node:internal/modules/cjs/loader:1005:19)
28 verbose stack     at require (node:internal/modules/cjs/helpers:102:18)
28 verbose stack     at Object.<anonymous> (C:\Users\Helldar\AppData\Roaming\nvm\v16.14.0\node_modules\npm\lib\utils\reify-output.js:16:38)
28 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1103:14)
29 verbose cwd C:\dev\domains\crm\frontend
30 verbose Windows_NT 10.0.19044
31 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "i"
32 verbose node v16.14.0
33 verbose npm  v8.3.1
34 error Unexpected token '.'
35 verbose exit 1
36 timing npm Completed in 357ms
37 verbose code 1
38 error A complete log of this run can be found in:
38 error     C:\Users\Helldar\AppData\Local\npm-cache\_logs\2022-02-14T13_39_55_054Z-debug-0.log
```

And I can't update the npm...

```
$ npm install -g npm@latest
npm ERR! Unexpected token '.'

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Helldar\AppData\Local\npm-cache\_logs\2022-02-14T13_44_21_865Z-debug-0.log
```

**Comment by DiracSpace** [comment]
> Node 16.14.0 Npm 8.3.1
> 
> Same error
> 
> ```
> 0 verbose cli [
> 0 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
> 0 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
> 0 verbose cli   'i'
> 0 verbose cli ]
> 1 info using npm@8.3.1
> 2 info using node@v16.14.0
> 3 timing npm:load:whichnode Completed in 1ms
> 4 timing config:load:defaults Completed in 2ms
> 5 timing config:load:file:C:\Users\Helldar\AppData\Roaming\nvm\v16.14.0\node_modules\npm\npmrc Completed in 3ms
> 6 timing config:load:builtin Completed in 4ms
> 7 timing config:load:cli Completed in 3ms
> 8 timing config:load:env Completed in 2ms
> 9 timing config:load:file:C:\dev\domains\crm\frontend\.npmrc Completed in 1ms
> 10 timing config:load:project Completed in 7ms
> 11 timing config:load:file:C:\Users\Helldar\.npmrc Completed in 1ms
> 12 timing config:load:user Completed in 2ms
> 13 timing config:load:file:C:\Program Files\nodejs\etc\npmrc Completed in 0ms
> 14 timing config:load:global Completed in 0ms
> 15 timing config:load:validate Completed in 2ms
> 16 timing config:load:credentials Completed in 0ms
> 17 timing config:load:setEnvs Completed in 1ms
> 18 timing config:load Completed in 25ms
> 19 timing npm:load:configload Completed in 25ms
> 20 timing npm:load:setTitle Completed in 0ms
> 21 timing config:load:flatten Completed in 3ms
> 22 timing npm:load:display Completed in 6ms
> 23 verbose logfile C:\Users\Helldar\AppData\Local\npm-cache\_logs\2022-02-14T13_39_55_054Z-debug-0.log
> 24 timing npm:load:logFile Completed in 14ms
> 25 timing npm:load:timers Completed in 0ms
> 26 timing npm:load:configScope Completed in 0ms
> 27 timing npm:load Completed in 46ms
> 28 verbose stack C:\Users\Helldar\AppData\Roaming\nvm\v16.14.0\node_modules\npm\node_modules\libnpmfund:1
> 28 verbose stack ../workspaces/libnpmfund
> 28 verbose stack ^
> 28 verbose stack
> 28 verbose stack SyntaxError: Unexpected token '.'
> 28 verbose stack     at Object.compileFunction (node:vm:352:18)
> 28 verbose stack     at wrapSafe (node:internal/modules/cjs/loader:1032:15)
> 28 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1067:27)
> 28 verbose stack     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
> 28 verbose stack     at Module.load (node:internal/modules/cjs/loader:981:32)
> 28 verbose stack     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
> 28 verbose stack     at Module.require (node:internal/modules/cjs/loader:1005:19)
> 28 verbose stack     at require (node:internal/modules/cjs/helpers:102:18)
> 28 verbose stack     at Object.<anonymous> (C:\Users\Helldar\AppData\Roaming\nvm\v16.14.0\node_modules\npm\lib\utils\reify-output.js:16:38)
> 28 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1103:14)
> 29 verbose cwd C:\dev\domains\crm\frontend
> 30 verbose Windows_NT 10.0.19044
> 31 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "i"
> 32 verbose node v16.14.0
> 33 verbose npm  v8.3.1
> 34 error Unexpected token '.'
> 35 verbose exit 1
> 36 timing npm Completed in 357ms
> 37 verbose code 1
> 38 error A complete log of this run can be found in:
> 38 error     C:\Users\Helldar\AppData\Local\npm-cache\_logs\2022-02-14T13_39_55_054Z-debug-0.log
> ```
> 
> And I can't update the npm...
> 
> ```
> $ npm install -g npm@latest
> npm ERR! Unexpected token '.'
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     C:\Users\Helldar\AppData\Local\npm-cache\_logs\2022-02-14T13_44_21_865Z-debug-0.log
> ```

Same, I can't even update npm version.

**Comment by andrey-helldar** [comment]
My solution to the problem is to download version 16.13.2 and install it.

**Comment by bartgriffith** [comment]
I had this issue with Node 16.14.0 and NPM 8.3.1.  But the real issue was with NVM.  I had NVM 1.1.7 but updated to the latest which is currently 1.1.9.  Then run   nvm uninstall 16.14.0  followed by   nvm install 16.14.0   and of course   nvm use 16.14.0.   
After that all was well.
https://github.com/coreybutler/nvm-windows/releases/tag/1.1.9

**Comment by kilotone** [comment]
I had this issue(using npm install) after going from node 17.3.0 to 17.5.0 and upgrading npm accordingly, also upgrading nvm from 1.1.7 to 1.1.9.
I ended up loading the command prompt as administrator, uninstalling 17.5.0 with nvm, then reinstalling and the issue was resolved.

**Comment by yamgarcia** [comment]
> My solution to the problem is to download version 16.13.2 and install it.

I see that error running "npm i". The way I found around that is similar. I used nvm to install node v17 with npm v8 and then use nvs to switch to 16 and then run "npm i" again.

**Comment by vigneshp200** [comment]
I found that the core problem is that nvm (in Windows) is not installing new node version properly.
You can manual add the node files into to the nvm node directories: C:\Users\{username}\AppData\Roaming\nvm\.
Go to the directory clear the contents node version folder (e.g. v16.14.0) and try installing node using msi or exe into the same folder. This time your node will be installed properly.

**Comment by gogbuehi** [comment]
> > > use the newest version of npm and u should be fine. just skip 8.3.1.
> > 
> > 
> > how to do that? to update npm you need npm.
> 
> Just ran in the same issue with node 16.14.0. You can download 8.3.2 [tgz file](https://registry.npmjs.org/npm/-/npm-8.3.2.tgz) and manually replace npm in your node_modules directory of the node version you have.

This worked for me. Manually creating an npm directory in my C:\Program Files\nodejs\node_modules\ directory resolved the issue.

**Comment by ljharb** [comment]
@vigneshp200 to be clear, nvm-windows is not the same thing as “nvm in windows”; they’re different projects.

**Comment by leafac** [comment]
To confirm what others are saying:

I ran into the same issue everyone is describing: `SyntaxError: Unexpected token '.'`

Node.js v17.6.0, npm v8.5.1

What fixed it for me was updating NVM for Windows from 1.1.7 to 1.1.9. Unfortunately, to do that I had to use the [installer](https://github.com/coreybutler/nvm-windows/tree/a7f30f5e122c88059e43971691ee177f6548e90f#installation--upgrades) by hand, since the latest version available through Chocolatey is 1.1.7. I opened https://github.com/coreybutler/nvm-windows/issues/757 to start working on this update.

**Comment by ljharb** [comment]
(to be clear, that project is “nvm-windows”; “nvm for windows” would just be real nvm on git bash or cygwin or WSL, all of which are supported)

**Comment by leafac** [comment]
@ljharb I’m sorry you have to keep clearing up this confusion for people who don’t know that [NVM for Windows](https://github.com/coreybutler/nvm-windows) isn’t the same as [nvm](https://github.com/nvm-sh/nvm). But “NVM for Windows” does present itself as “NVM for Windows” (look at their [README](https://github.com/coreybutler/nvm-windows/tree/a7f30f5e122c88059e43971691ee177f6548e90f)), so I believe it’s appropriate to refer to it as “NVM for Windows”.

Maybe you and @coreybutler should talk about this (at some other place, presumably)?

**Comment by ljharb** [comment]
Fair enough; filed https://github.com/coreybutler/nvm-windows/issues/758

**Comment by timothystewart6** [comment]
> I am using the latest `nvm-windows` and I cannot install any packages using node `16.14.0` with npm `8.3.1` which was installed via `nvm-windows`
> 
> ```
> npm install -g pm2
> npm ERR! Unexpected token '.'
> ```

Just confirming that upgrading `nvm for Windows` (Windows installer, not WSL) did fix my issue but only after uninstalling the existing version and cleaning up the previous install.

**Comment by leafac** [comment]
Follow up for Chocolatey users out there: An update to NVM for Windows [is on the way](https://community.chocolatey.org/packages/nvm/1.1.9#comment-5746314566) and you may get it now with:

```console
> choco install nvm --version=1.1.9
```

**Comment by daveamol** [comment]
Seems like this is broken for nvm-sh (version 1.1.7, installed from https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh) when using cmder with bash. Using 16.13.2 works for now

**Comment by ljharb** [comment]
@daveamol thats a confusing comment, because nvm-sh has never released anything v1 or above; those versions apply to nvm-windows.

**Comment by daveamol** [comment]
Ignore what I said, I ran the wrong script thinking it would update my nvm-windows and my version is still stuck on 1.1.7

**Comment by UziTech** [comment]
Can this issue be re opened? or would it be better to create a new issue? This is still a problem.

**Comment by RodEsp** [comment]
> Can this issue be re opened? or would it be better to create a new issue? This is still a problem.

@UziTech this issue isn't a problem with NPM. It's a problem with [`nvm-windows`](https://github.com/coreybutler/nvm-windows) version `1.1.7`. Updating to to the latest `nvm-windows` and reinstalling node/npm with it will fix the problem. Check out upgrade instructions on the `nvm-windows` repo.

**Comment by DiracSpace** [comment]
> Updating to to the latest nvm-windows and reinstalling node/npm with it will fix the problem.

I can confirm that

**Comment by WORMSS** [comment]
Strange, I just updated to 1.1.9 of [`nvm-windows`](https://github.com/coreybutler/nvm-windows) and I still get the same problem.
Though,
```ts
nvm install 17.1.0
nvm use 17.1.0 (this also changes to npm 8.3.1)
npm i -D @types/node
nvm use 17.6.0 (this also changes to npm 8.5.1)
```
Worked perfectly for my needs.. Run node with 17.6, but install things with 17.1..

**Comment by timhunt** [comment]
Is there someone here with enough permissions to edit the title of this issue?

I think editing it to 

npm ERR! Unexpected token '.' with nvm-windows 1.1.7

would help people arriving at this issue for the first time (like I just did) to find the solution quicker. Thanks.

**Comment by MylesBorins** [comment]
done

**Comment by timhunt** [comment]
Thank you @MylesBorins :-)

**Comment by Daalb** [comment]
> I had this issue with Node 16.14.0 and NPM 8.3.1. But the real issue was with NVM. I had NVM 1.1.7 but updated to the latest which is currently 1.1.9. Then run nvm uninstall 16.14.0 followed by nvm install 16.14.0 and of course nvm use 16.14.0. After that all was well. https://github.com/coreybutler/nvm-windows/releases/tag/1.1.9

This worked for me. Thanks!!

**Comment by Daalb** [comment]
> I had this issue with Node 16.14.0 and NPM 8.3.1. But the real issue was with NVM. I had NVM 1.1.7 but updated to the latest which is currently 1.1.9. Then run nvm uninstall 16.14.0 followed by nvm install 16.14.0 and of course nvm use 16.14.0. After that all was well. https://github.com/coreybutler/nvm-windows/releases/tag/1.1.9

This worked for me. Thank you @bartgriffith 😄

**Comment by LeoEatle** [comment]
> I had this issue with Node 16.14.0 and NPM 8.3.1. But the real issue was with NVM. I had NVM 1.1.7 but updated to the latest which is currently 1.1.9. Then run nvm uninstall 16.14.0 followed by nvm install 16.14.0 and of course nvm use 16.14.0. After that all was well. https://github.com/coreybutler/nvm-windows/releases/tag/1.1.9

This works for me. Thanks!

**Comment by TramVu** [comment]
> I had this issue(using npm install) after going from node 17.3.0 to 17.5.0 and upgrading npm accordingly, also upgrading nvm from 1.1.7 to 1.1.9. I ended up loading the command prompt as administrator, uninstalling 17.5.0 with nvm, then reinstalling and the issue was resolved.

Hi how can you upgrade the nvm version please ?

**Comment by jianbin-lim** [comment]
> > I had this issue with Node 16.14.0 and NPM 8.3.1. But the real issue was with NVM. I had NVM 1.1.7 but updated to the latest which is currently 1.1.9. Then run nvm uninstall 16.14.0 followed by nvm install 16.14.0 and of course nvm use 16.14.0. After that all was well. https://github.com/coreybutler/nvm-windows/releases/tag/1.1.9
> 
> This works for me. Thanks!

>  This works for me too !
1. update the nvm to 1.1.9
2. nvm uninstall 16.14.2
3. nvm install 16.14.2 and nvm use 16.14.2

**Comment by oktayibis** [comment]
The tricky part is that you need to update the nvm before the install/update node. If you already installed the node before updating nvm, delete it with nvm and make sure deleted folder is named node version in the nvm folder.
 and follow the above instructions 
 
So confirmed that also this works.
  [@jianbin](https://github.com/npm/cli/issues/4234#issuecomment-1080927985)

**Comment by yosmarcode** [comment]
Este error igual sale en la version de nodejs nvm - version 16.14.2

**Comment by bombillazo** [comment]
> Este error igual sale en la version de nodejs nvm - version 16.14.2

Mira este comentario. https://github.com/npm/cli/issues/4234#issuecomment-1040528646

**Comment by yosmarcode** [comment]
> > Este error igual sale en la version de nodejs nvm - version 16.14.2
> 
> Mira este comentario. [#4234 (comentario)](https://github.com/npm/cli/issues/4234#issuecomment-1040528646)

Si gracias amigo lo pude solventar el problema principal era que la version de nvm estaba presentando inconvenientes pero realizar un update a la ultima version 1.1.9 y desinstalé las versiones de nodejs que tenia instalada e  instalar de nuevo
Y todo volvió a la normalidad

gracias

**Comment by rajugood** [comment]
faced same issue when i use `node v16.14.2`, and tried `npx create-react-app my-app`

Solved after switching to different version of node.

**Comment by ghost** [comment]
well, after change nvm version from 1.1.7 to 1.1.7, using node 16.14.2 I still have this problem.

**Comment by bartgriffith** [comment]
> well, after change nvm version from 1.1.7 to 1.1.7, using node 16.14.2 I still have this problem.

I assume you mean 1.1.7 to 1.1.9.
If you have installed NVM 1.1.9, uninstall 16.14.2 and then reinstall using NVM.
nvm uninstall 16.14.2
nvm install 16.14.2
nvm use 16.14.2
Good Luck!

**Comment by ghost** [comment]
> > well, after change nvm version from 1.1.7 to 1.1.7, using node 16.14.2 I still have this problem.
> 
> I assume you mean 1.1.7 to 1.1.9. If you have installed NVM 1.1.9, uninstall 16.14.2 and then reinstall using NVM. nvm uninstall 16.14.2 nvm install 16.14.2 nvm use 16.14.2 Good Luck!

Oh right, it was that was left to do, now is working properly, thanks.

**Comment by yosmarcode** [comment]
Hola es que pueden solventar el problema es desintarla la version que tienes actual de nvm 1.1.7, e installar version 1.1.9 ya que ella solventan el inconveniente y alli procedes a instalar la version mas reciente nvm install node latest, y cuando tengas esa version estables, 

Poder instlar la version que necesitas ocupar

nvm use vx.xxx

y listo

espero que te pueda ayudar

**Comment by akshayjai1** [comment]
Hello, is that they can solve the problem is to uninstall the current version of nvm 1.1.7, and install version 1.1.9 since it solves the problem and there you proceed to install the most recent version nvm install node latest, and when you have that stable version,

To be able to install the version you need to occupy

nvm use vx.xxx

and ready

I hope it will help you

**Comment by clementmas** [comment]
> 1. update the nvm to 1.1.9
> 2. nvm uninstall 16.14.2
> 3. nvm install 16.14.2 and nvm use 16.14.2

This worked for me. Note for step 1: I couldn't do it from the console on Windows. I had to download the `nvm-update.zip` file from the latest release

**Comment by WORMSS** [comment]
> > 1. update the nvm to 1.1.9
> > 2. nvm uninstall 16.14.2
> > 3. nvm install 16.14.2 and nvm use 16.14.2
> 
> This worked for me. Note for step 1: I couldn't do it from the console on Windows. I had to download the `nvm-update.zip` file from the latest release

That doesn't work for me, because I get
1. Update works fine.
2. nvm uninstall 16.3.0 `// Uninstalling node v16.3.0...exit status 5: Access is denied.`

**Comment by bartgriffith** [comment]
> > > 1. update the nvm to 1.1.9
> > > 2. nvm uninstall 16.14.2
> > > 3. nvm install 16.14.2 and nvm use 16.14.2
> > 
> > 
> > This worked for me. Note for step 1: I couldn't do it from the console on Windows. I had to download the `nvm-update.zip` file from the latest release
> 
> That doesn't work for me, because I get
> 
> 1. Update works fine.
> 2. nvm uninstall 16.3.0 `// Uninstalling node v16.3.0...exit status 5: Access is denied.`

Try running your console/terminal as administrator.

**Comment by WORMSS** [comment]
That is not a solution. That is a work-around.. 
This problem requires a solution.

**Comment by yosmarcode** [comment]
Otra solución sería, realizar los paso anteriormente mencionados, en diferentes comentarios

Cuando realicen la instalación de nvm.
Habrán la terminal de su sistema operativo con privilegio de administrador

Luego procesadan a realizar las instalación de nodejs más actual

Cuando esto suceda ya pueden instalar la versión que más le acomoda

Esto igual me resultó

**Comment by DaviWeiss** [comment]
> I had a similar issue when using `npm init` and `npm install`. I downgrade my node version to 16.13.2 (recommended for most users) and it works.

This worked for me too

**Comment by snake-py** [comment]
I am still seeing this on win 10:

nvm 1.1.7
Node 16.5 and 18.0
npm 8.5.5 and 8.6

could only resolve once I downgraded to node 14.18.3 and npm 6.14.15

**Comment by yosmarcode** [comment]
> Todavía estoy viendo esto en win 10:
> 
> nvm 1.1.7 Nodo 16.5 y 18.0 npm 8.5.5 y 8.6
> 
> solo pude resolver una vez que bajé al nodo 14.18.3 y npm 6.14.15

Trata de actualizar la versión de nvm 1.1.9

**Comment by WORMSS** [comment]
nvm 1.1.9 has access denied problems.

```sh
exit status 1: Access is denied.
```
The way to get around this issue is to keep with nvm 1.1.7

**Comment by yosmarcode** [comment]
> nvm 1.1.9 has access denied problems.
> 
> ```shell
> exit status 1: Access is denied.
> ```
> 
> The way to get around this issue is to keep with nvm 1.1.7


No es necesario solo tratar de iniciar la teminar en modo administrador  eso dependiendo del sistema operativo en uso

eso es solo una vez 

espero que te sea util

**Comment by WORMSS** [comment]
> > nvm 1.1.9 has access denied problems.
> > ```shell
> > exit status 1: Access is denied.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The way to get around this issue is to keep with nvm 1.1.7
> 
> No es necesario solo tratar de iniciar la teminar en modo administrador eso dependiendo del sistema operativo en uso
> 
> eso es solo una vez
> 
> espero que te sea util

Can't do it if you are not an admin.

**Comment by silverkorn** [comment]
You can also make it like it was on nvm 1.1.7 and popup for adminstrator rights using Powershell:
```batch
powershell "Start-Process -FilePath 'nvm.exe' -ArgumentList 'use 14.18.3' -Verb runAs"
```

**Comment by asjensse** [comment]
Is this solved in the newest version `Node.js v18.0.0`?
Or is it still recommended to use `Node.js v16.X.X`?

**Comment by cristiand391** [comment]
@asjensse the issue is caused by a change in npm, it only happens when using `nvm-windows` **on Windows** see: https://github.com/npm/cli/issues/4340#issuecomment-1035608510

They aren't going to revert that, so the recommended fix is to use nvm-windows v1.1.8, you can pin node to v16.13.2 which is the last known "good" version if your pipelines are blocked.

If you aren't using Windows and `nvm-windows` then you should open another issue.

**Comment by Vampire2008** [comment]
This bug also exists on Node 14.19.2 + npm 8.10.0

**Comment by hungnc051220** [comment]
> > I had this issue with Node 16.14.0 and NPM 8.3.1. But the real issue was with NVM. I had NVM 1.1.7 but updated to the latest which is currently 1.1.9. Then run nvm uninstall 16.14.0 followed by nvm install 16.14.0 and of course nvm use 16.14.0. After that all was well. https://github.com/coreybutler/nvm-windows/releases/tag/1.1.9
> 
> This works for me. Thanks!

This worked, thank you so much!

**Comment by mcaden** [comment]
I found I was able to use latest `nvm` and LTS of Node just fine without downgrading. Both cases I ran into this on different computers running different versions of NVM. I installed `16.15.0` and then hit this exact bug. But rather than downgrading:

1) Install nvm 1.1.9
2) `nvm uninstall 16.15.0`
3) `nvm install 16.15.0`
4) `nvm use 16.15.0`

Simply reinstalling and using while have the latest issue resolved it.

**Comment by afgm05** [comment]
Uninstalling node.js and nvm and reinstalling them again seems to fix the issue for me.

**Comment by HarrySpyrou** [comment]
Apologies if I'm hijacking the thread but I got 18.6 working for me with nvm 1.1.7 on Windows 10:

`nvm uninstall 18.6` (if you have it installed).

I went to the [node.js website and downloaded the Windows msi ](https://nodejs.org/en/download/current/) and when it asked me if it should install 'additional tools' I selected that (which I believe is what was going wrong with nvm). Then I let it do its thing, took a while, installed its whole family and then some.

then `nvm install 18.6` and worked just fine.

I think nvm does not install all of what's needed (or maybe doesn't install it properly).

**Comment by srsgores** [comment]
As @mcaden wrote, Windows users must re-install the `latest` version of NodeJS using `nvm`:

1. Download and re-run the latest [`nvm-setup.exe`](https://github.com/coreybutler/nvm-windows/releases/download/1.1.9/nvm-setup.exe) from the [`nvm-windows` release page](https://github.com/coreybutler/nvm-windows/releases)
2. Uninstall the `latest` version of NodeJS via a Terminal session with Administrator privileges:
```bash
nvm uninstall latest
```
3. Re-install the latest version of NodeJS
```bash
nvm inistall latest
```
4. **Use** the `latest` version of NodeJS
```bash
nvm use latest
```
Now you can try re-installing a package globally to verify that this has worked and has taken effect:

```bash
node --version
> Now using node v18.6.0 (64-bit)
```

**Comment by WORMSS** [comment]
The problem here is _"with Administrator privileges"._. 
We **SHOULD** get the Windows UAC screen (looks like)
![image](https://user-images.githubusercontent.com/1384537/179915264-376ba44b-865b-42d8-a975-2ce40405d646.png)

Where you allow/disallow each individual command.
Rather than open up "with administrator privileges" where you give it permission to do anything and everything it wants.. 

It's a stupid situation where you need "**1.1.8 or higher** to install latest node" but you need "**1.1.7 or lower** to use nvm on a day to day bases.."

**Comment by bartgriffith** [comment]
I am using NVM 1.1.9 and don't have any issues installing and switching between node versions. In fact, I installed 16.16.0 this morning.

I think the issue is trying to use versions of node that were installed with previous versions of NVM.  When you upgrade NVM you will have to uninstall and reinstall all the versions of node with the latest NVM or you will see this error.

**Comment by WORMSS** [comment]
Sadly it was not the case, 1.1.8 and higher produces a `Access is denied` and doesn't attempt to bring up the UAC screen at all.

**Comment by HarrySpyrou** [comment]
@bartgriffith what about Node 11 or 10?

**Comment by AJMcKane** [comment]
Still reproducing on Node 17.7.2 with NPM 8.5.2 installed via NVM on Windows 10.  Also had the same issue with Node 16.16.0 and NPM 8.11.0.  Had to revert  to Node v16.13.2  NPM 8.1.2 to get `npm install` to work without this error.

**Comment by cristiand391** [comment]
@AJMcKane this is not an npm bug, it only happens with older nvm version, read the prev comments.

**Comment by j0eii** [comment]
still hitting this issue.
node v16.16.0
npm  v8.11.0
nvm 1.1.9

**Comment by j0eii** [comment]
> 16.13.2

^^ it works

**Comment by satyamdube** [comment]
> I had a similar issue when using `npm init` and `npm install`. I downgrade my node version to 16.13.2 (recommended for most users) and it works.

working fine thanks

**Comment by Kokuyo3** [comment]
I was seeing this same ERR message when running create-react-app 5.0.1 through Intellij on nvm-windows 1.1.7 After removing node then updating nvm-windows to 1.1.9 and re-installing node, I no longer see the issue with the following setup: 

- node 16.16.0 (64 bit)
- npm 8.11.0
- nvm-windows 1.1.9

**Comment by TinaSprunt** [comment]
> Yesterday node v16.14.0 (latest LTS) was released, and it includes npm v8.3.1 so there's no way to make npm update itself. If you see this error when running latest LTS on windows, you might want to install previous LTS v16.13.2 👍🏼

Thank you for saving me

**Comment by jjmato** [comment]
same issue 

nvm    1.1.7
node  16.16.0
npm   6.11.0


```
0 verbose cli C:\nodejs\node.exe C:\nodejs\node_modules\npm\bin\npm-cli.js                                                                          
1 info using npm@8.11.0                                                                                                                             
2 info using node@v16.16.0                                                                                                                          
3 timing npm:load:whichnode Completed in 0ms                                                                                                        
4 timing config:load:defaults Completed in 2ms                                                                                                      
5 timing config:load:file:C:\nvm\v16.16.0\node_modules\npm\npmrc Completed in 3ms                                                                   
6 timing config:load:builtin Completed in 3ms                                                                                                       
7 warn config global `--global`, `--local` are deprecated. Use `--location=global` instead.                                                         
8 timing config:load:cli Completed in 4ms                                                                                                           
9 timing config:load:env Completed in 3ms                                                                                                           
10 timing config:load:project Completed in 4ms                                                                                                      
11 timing config:load:file:C:\Users\username\.npmrc Completed in 2ms                                                                                  
12 timing config:load:user Completed in 3ms                                                                                                         
13 timing config:load:file:C:\nodejs\etc\npmrc Completed in 1ms                                                                                     
14 timing config:load:global Completed in 1ms                                                                                                       
15 timing config:load:validate Completed in 2ms                                                                                                     
16 timing config:load:credentials Completed in 1ms                                                                                                  
17 timing config:load:setEnvs Completed in 1ms                                                                                                      
18 timing config:load Completed in 25ms                                                                                                             
19 timing npm:load:configload Completed in 25ms                                                                                                     
20 timing npm:load:mkdirpcache Completed in 2ms                                                                                                     
21 timing npm:load:mkdirplogs Completed in 1ms                                                                                                      
22 verbose title npm i npm                                                                                                                          
23 verbose argv "i" "--global" "npm"                                                                                                                
24 timing npm:load:setTitle Completed in 2ms                                                                                                        
25 timing config:load:flatten Completed in 5ms                                                                                                      
26 timing npm:load:display Completed in 10ms                                                                                                        
27 verbose logfile logs-max:10 dir:C:\nodejs\npm-cache\_logs                                                                                        
28 verbose logfile C:\nodejs\npm-cache\_logs\2022-08-11T11_37_16_597Z-debug-0.log                                                                   
29 timing npm:load:logFile Completed in 8ms                                                                                                         
30 timing npm:load:timers Completed in 0ms                                                                                                          
31 timing npm:load:configScope Completed in 0ms                                                                                                     
32 timing npm:load Completed in 51ms                                                                                                                
33 verbose stack C:\nvm\v16.16.0\node_modules\npm\node_modules\libnpmfund:1                                                                         
33 verbose stack ../workspaces/libnpmfund                                                                                                           
33 verbose stack ^                                                                                                                                  
33 verbose stack                                                                                                                                    
33 verbose stack SyntaxError: Unexpected token '.'                                                                                                  
33 verbose stack     at Object.compileFunction (node:vm:352:18)                                                                                     
33 verbose stack     at wrapSafe (node:internal/modules/cjs/loader:1033:15)                                                                         
33 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1069:27)                                                                  
33 verbose stack     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)                                                    
33 verbose stack     at Module.load (node:internal/modules/cjs/loader:981:32)                                                                       
33 verbose stack     at Function.Module._load (node:internal/modules/cjs/loader:822:12)                                                             
33 verbose stack     at Module.require (node:internal/modules/cjs/loader:1005:19)                                                                   
33 verbose stack     at require (node:internal/modules/cjs/helpers:102:18)                                                                          
33 verbose stack     at Object.<anonymous> (C:\nvm\v16.16.0\node_modules\npm\lib\utils\reify-output.js:16:38)                                       
33 verbose stack     at Module._compile (node:internal/modules/cjs/loader:1105:14)                                                                  
34 verbose cwd D:\@workspace                                                                               
35 verbose Windows_NT 10.0.18363                                                                                                                    
36 verbose node v16.16.0                                                                                                                            
37 verbose npm  v8.11.0                                                                                                                             
38 error Unexpected token '.'                                                                                                                       
39 verbose exit 1                                                                                                                                   
40 timing npm Completed in 79ms                                                                                                                     
41 verbose code 1                                                                                                                                   
42 error A complete log of this run can be found in:                                                                                                
42 error     C:\nodejs\npm-cache\_logs\2022-08-11T11_37_16_597Z-debug-0.log   
```

**Comment by michaelprescott** [comment]
Same issue:

node v18.7.0
npm 8.15.0
nvm 1.1.7

To avoid the problem for the moment, things work at:
node v14.18.1
npm 6.14.15

**Comment by cristian495** [comment]
i faced the same problem with Node 16.16.0, the solution was downgrade the Node version to 16.13.2

Commands
nvm install 16.13.2
nvm use 16.13.2

**Comment by matheussbandeira** [comment]
i faced the same problem with Node 16.17.0, the solution was downgrade the Node version to 16.13.2

Commands
nvm install 16.13.2
nvm use 16.13.2

**Comment by S-amuel** [comment]
Issue leading me to this thread on:
Node: v18.7.0
NPM: v8.15.0

**Comment by twittwer** [comment]
Same issue with node's latest LTS on windows
node: 16.17.0
npm: 8.15.0

**Comment by anderson012** [comment]
@twittwe here too :'(
node: 16.17.0 on windows
npm: 8.15.0

**Comment by imera92** [comment]
these are the versions that led me to this thread:
nvm: 1.1.5
node: 16.14.0
npm: 8.3.1

I was getting that nasty "Unexpected token '.'" error when trying to install, check version, and using npx (which was my main concern since I needed it for a script). I then tried switching versions:

nvm: 1.1.5
node: 16.17.0
npm: 8.15.0

I was still getting that error, so I did something similar to what @srsgores mentioned (all from a PowerShell command prompt running as Administrator):

1) updated nvm to 1.1.9 (did this using chocolatey, not using the .exe installer)
2) uninstalled both 16.14.0 and 16.17.0 versions of node
3) reinstalled 16.17.0 version of node (latest LTS release at the moment of writing this comment)

After that, everything was working correctly again. My guess is nvm's version was the culprit, but I'm not sure. I hope this is of use to someone.

**Comment by TGTGamer** [comment]
> After that, everything was working correctly again. My guess is nvm's version was the culprit, but I'm not sure. I hope this is of use to someone.

Same issue, different versions. Followed the steps @imera92 suggests and it was resolved. Working solution:

1. Uninstall nvm and reinstall (or `choco upgrade nvm`)
2. Uninstall all versions of node (`nvm uninstall <node version>`)
3. Reinstall needed node version (`nvm install <node version>`)
4. Use node version just installed (`nvm use <node version>`)

**Comment by liviublidar** [comment]
Why is this closed? any resolution? Issue still exists

**Comment by lambacini** [comment]
i faced the same problem when using windows and nvm. update nvm the new version, delete and reinstall the node version you want to use. (nvm uninstall version & nvm install version). this solved my problem

**Comment by ddomen** [comment]
Under Windows environments, the problem is related to symbolic links that are wrongly created as hard files.
For example:  `nodejs/node_modules/npm/node_modules/@npmcli/arborist` is a text file which contains the text `../../workspace/arborist`, but should be a folder symbolic link to that folder instead (`nodejs/node_modules/npm/node_modules/workspace/arborist`).

The node which run npm-cli indeed states that there is an _unexpected token '.'_, which is the content of the file arborist (`../../workspace/arborist`) which is read as a .js file instead of a link.

While waiting for an official resolution, I've fixed it by scanning the npm node_modules folder and fixing the links, here a node.js script to automate it:
https://gist.github.com/ddomen/ac17ef3edb15712130c3db3d47e4b377

After running the script you can use `npm i -g npm@latest` to restore the correct node environment (correctly resolving the packages instead of creating symbolic links to them)

**Comment by piotr-kubica** [comment]
Similar issue for v16.17.0.

**Comment by micke-dahlgren** [comment]
> Why is this closed? any resolution? Issue still exists

This issue was fixed with nvm 1.1.9. See @TGTGamer reply for instructions.

**Comment by nelsson** [comment]
> > Why is this closed? any resolution? Issue still exists
> 
> This issue was fixed with nvm 1.1.9. See @TGTGamer reply for instructions.

It works!

**Comment by pillesy** [comment]
Upgrading to nvm 1.1.9. did not immediately fix the issue for me, I had to uninstall and reinstall the broken node version.

- Uninstall broken versions of node (nvm uninstall <node version>)
- Reinstall needed node version (nvm install <node version>)
- Use node version just installed (nvm use <node version>)

**Comment by micke-dahlgren** [comment]
> Upgrading to nvm 1.1.9. did not immediately fix the issue for me, I had to uninstall and reinstall the broken node version.
> 
> * Uninstall broken versions of node (nvm uninstall )
> * Reinstall needed node version (nvm install )
> * Use node version just installed (nvm use )
@TGTGamer already outlined this. :)
https://github.com/npm/cli/issues/4234#issuecomment-1236659792

**Comment by micke-dahlgren** [comment]
@TGTGamer already outlined this. :)
https://github.com/npm/cli/issues/4234#issuecomment-1236659792

**Comment by jkupcho** [comment]
I'm also having this issue, it seems like this problem started with version 8.3.1.

**Comment by arcreigh** [comment]
Can confirm in current NPM build this is a problem.

**Comment by WORMSS** [comment]
NVM Version 1.1.10  has all these problems (plus the elevation problem) fixed.
Upgrade nvm, and then remove the broken node versions and re-install them.

Everything now works as expected.

**Comment by josefosaurus** [comment]
in my case I fix it uninstalling nvm and all versions of node and install everything again via chocolatey, nvm (latest version) and node 16.13.0  and works as expected.

**Comment by SensSebastianSchlosser** [comment]
> NVM Version 1.1.10 has all these problems (plus the elevation problem) fixed. Upgrade nvm, and then remove the broken node versions and re-install them.
> 
> Everything now works as expected.


This is the solution!

**Comment by aoneahsan** [comment]
> these are the versions that led me to this thread: nvm: 1.1.5 node: 16.14.0 npm: 8.3.1
> 
> I was getting that nasty "Unexpected token '.'" error when trying to install, check version, and using npx (which was my main concern since I needed it for a script). I then tried switching versions:
> 
> nvm: 1.1.5 node: 16.17.0 npm: 8.15.0
> 
> I was still getting that error, so I did something similar to what @srsgores mentioned (all from a PowerShell command prompt running as Administrator):
> 
> 1. updated nvm to 1.1.9 (did this using chocolatey, not using the .exe installer)
> 2. uninstalled both 16.14.0 and 16.17.0 versions of node
> 3. reinstalled 16.17.0 version of node (latest LTS release at the moment of writing this comment)
> 
> After that, everything was working correctly again. My guess is nvm's version was the culprit, but I'm not sure. I hope this is of use to someone.

This resolved the issue for me, uninstalled the "nvm" installed using ".exe" file and reinstall using "chocolatey"
and then install the node again and use that it will work (at least worked for me :)

**Comment by sleepinzombie** [comment]
The above works for me as well.
Thank you @aoneahsan!

**Comment by LucasHiago** [comment]
> Fixed it on my own. Please replace it with my version https://files.fm/u/sz3yrgsvj

vai tomar no seu cu irmão vai deixar o link vazio, tu que se foda. ajuda de merda em.

**Comment by mohammadmoth** [comment]
> > these are the versions that led me to this thread: nvm: 1.1.5 node: 16.14.0 npm: 8.3.1
> > I was getting that nasty "Unexpected token '.'" error when trying to install, check version, and using npx (which was my main concern since I needed it for a script). I then tried switching versions:
> > nvm: 1.1.5 node: 16.17.0 npm: 8.15.0
> > I was still getting that error, so I did something similar to what @srsgores mentioned (all from a PowerShell command prompt running as Administrator):
> > 
> > 1. updated nvm to 1.1.9 (did this using chocolatey, not using the .exe installer)
> > 2. uninstalled both 16.14.0 and 16.17.0 versions of node
> > 3. reinstalled 16.17.0 version of node (latest LTS release at the moment of writing this comment)
> > 
> > After that, everything was working correctly again. My guess is nvm's version was the culprit, but I'm not sure. I hope this is of use to someone.
> 
> This resolved the issue for me, uninstalled the "nvm" installed using ".exe" file and reinstall using "chocolatey" and then install the node again and use that it will work (at least worked for me :)




I deleted all versions of node 18, and then re-downloaded it, it works.

**Comment by webdevatanas** [comment]
I am using windows and here is how I fixed it:
1. Uninstall nvm from windows(you can do that from **Add and remove programs**). This will remove also the node versions installed with the nvm previously so you do not need manually to uninstall all before uninstall the whole nvm.
2. Install the latest nvm from here https://github.com/coreybutler/nvm-windows/tags . 1.1.10 is the latest to this date.
3. nvm install 'your prefered version' then nvm use 'the installed version'
4. Whoala!

**Comment by dosstx** [comment]
> I am using windows and here is how I fixed it:
> 
> 1. Uninstall nvm from windows(you can do that from **Add and remove programs**). This will remove also the node versions installed with the nvm previously so you do not need manually to uninstall all before uninstall the whole nvm.
> 2. Install the latest nvm from here https://github.com/coreybutler/nvm-windows/tags . 1.1.10 is the latest to this date.
> 3. nvm install 'your prefered version' then nvm use 'the installed version'
> 4. Whoala!

Fantastic step by step. THank you!

**Comment by maurymmarques** [comment]
I've updated my node version and it worked.

Linux/macOS:

1. Open the Terminal and check your current Node version: 
    `node -v`
2. Install n package using the following command:
    `npm install -g n`
This command will install a tool called "n" which you can use to update Node easily.
3. To update Node, run the following command in your terminal:           
    `n latest`
This command will install the latest version of Node on your system.
4. Now you can verify that your update is complete by rechecking your Node version:  
    `node -v`

**Comment by coreybutler** [comment]
@maurymmarques - n is a tool for _Linux/macOS_, not for Windows.

**Comment by natesire** [comment]
If npm isn't working... then how can I use npm to fix itself. It's a circular problem. I cannot run any npm commands.

**Comment by natesire** [comment]
The easiest way to work around this is to either use Linux or the Ubuntu Subsystem on Windows to run your npm commands. And run with --verbose so you can see that it's the funding package libnpmfund that is tripping up over a path syntax error with the '.' on Windows.

**Comment by natesire** [comment]
It is happening with random npm packages. So that points to a problem with the npm code that calls out to these packages.

at Module._compile (node:internal/modules/cjs/loader:1159:14

**Comment by natesire** [comment]
I found the root cause. These paths inside these files that the loader is requiring need double quoted. They are regular expressions too.

Change 
../workspaces/xxx

To
"../workspaces/xxx"

in each file and it will work on Windows. I don't know why this is only breaking on Windows.

**Comment by natesire** [comment]
https://github.com/npm/cli/pull/6601

**Comment by Tan-nn** [comment]
I faced issues with nvm 1.1.7, Solved this issue when updated to 1.1.12, but I can not uninstall nvm 1.1.7 only copy and replace file

**Comment by carlosananias** [comment]
> A versão 1.1.10 do NVM tem todos esses problemas (mais o problema de elevação) corrigidos. Atualize o nvm e, em seguida, remova as versões do nó quebrado e reinstale-as.
> 
> Agora tudo funciona conforme o esperado.

this is excellent... works well for my team

**Comment by carlosananias** [comment]
> > Depois disso, tudo voltou a funcionar corretamente. Meu palpite é que a versão do nvm foi a culpada, mas não tenho certeza. Espero que isso seja útil para alguém.
> 
> Mesmo problema, versões diferentes. Segui os passos@imera92sugere e foi resolvido. Solução de trabalho:
> 
> 1. Desinstale o nvm e reinstale (ou `choco upgrade nvm`)
> 2. Desinstale todas as versões do node ( `nvm uninstall <node version>`)
> 3. Reinstale a versão necessária do nó ( `nvm install <node version>`)
> 4. Use a versão do nó recém-instalada ( `nvm use <node version>`)

this is excellent... works well for my team
